### PR TITLE
Add a Makefile with easy to discover targets

### PR DESCRIPTION
**Pull request by viovanov**
## Description
A makefile to call bazel and other useful commands

## Motivation and Context
It's hard to discover bazel targets, this can make it easier
See #545 

## How Has This Been Tested?
I use this all the time locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	bazel run //dev/kubecf:apply
+
+# Installs the CF operator (TODO: this should be done with bazel)
+operatorup:
+	k create namespace cf-operator && \
+	bazel run //dev/cf_operator:apply
+
+# Cleans everything up, but doesn't destroy your cluster
+# (TODO: this should be done with bazel)
+clean:
```
Comment (mudler) [inline Makefile]: It implies `k` is installed on the host, which would break our "user contract", no?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
+# Destroys your kind cluster completely
+kinddown:
```
Comment (mudler) [inline Makefile]: Just my 2c on this: we should scope a bit `bazel` usage to just build, and not try to make it a deploy platform. Feel free to ignore this comment :)

**Comment by mudler** [comment]
I'm a bit against a `Makefile` here, already expressed my thoughts on https://github.com/cloudfoundry-incubator/kubecf/issues/545 and I won't repeat them here :smiley: 

I think at least the Makefile should have `dependency` or `deps` target to get `bazel` on the first place

File: Makefile (hunk 1 above)
Comment (viovanov) [reply Makefile]: sure, i missed that

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+
+# Destroys your kind cluster completely
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
# Destroys your kind cluster completely.
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+kinddown:
+	bazel run //dev/kind:delete
+
+# Creates a new kind cluster for you
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
# Creates a new kind cluster for you.
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+kindup:
+	bazel run //dev/kind:start
+
+# Lists all bazel targets, so you can grep for what you need
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
# Lists all bazel targets, so you can grep for what you need.
```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+bazel-targets:
+	bazel query 'attr(visibility, "//visibility:public", ...)' && \
+	bazel query 'attr(visibility, "", ...)'
+
+# Installs/updates KubeCF (on whatever cluster your kubectl points to)
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
# Installs/updates KubeCF (on whatever cluster your kubectl points to).
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+apply:
+	bazel run //dev/kubecf:apply
+
+# Installs the CF operator (TODO: this should be done with bazel)
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
# Installs the CF operator.
# TODO: this should be done with bazel.
```

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+operatorup:
+	k create namespace cf-operator && \
+	bazel run //dev/cf_operator:apply
+
+# Cleans everything up, but doesn't destroy your cluster
+# (TODO: this should be done with bazel)
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
# Cleans everything up but doesn't destroy your cluster.
# TODO: this should be done with bazel.
```

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+	k delete namespace cf-operator ; \
+	k delete clusterrole cf-operator-quarks-job ; \
+	k delete clusterrole cf-operator ; \
+	k delete crd boshdeployments.quarks.cloudfoundry.org ; \
+	k delete crd quarksjobs.quarks.cloudfoundry.org ; \
+	k delete crd quarkssecrets.quarks.cloudfoundry.org ; \
+	k delete crd quarksstatefulsets.quarks.cloudfoundry.org ; \
+	k delete clusterrolebinding cf-operator-quarks-job ; \
+	k delete clusterrolebinding cf-operator ; \
+	k delete psp kubecf-default
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
	bazel run @helm//:binary -- delete kubecf ; \
	bazel run @helm//:binary -- delete cf-operator ; \
	bazel run @kubectl//:binary -- delete namespace kubecf ; \
	bazel run @kubectl//:binary -- delete namespace cf-operator ; \
	bazel run @kubectl//:binary -- delete clusterrole cf-operator-quarks-job ; \
	bazel run @kubectl//:binary -- delete clusterrole cf-operator ; \
	bazel run @kubectl//:binary -- delete crd boshdeployments.quarks.cloudfoundry.org ; \
	bazel run @kubectl//:binary -- delete crd quarksjobs.quarks.cloudfoundry.org ; \
	bazel run @kubectl//:binary -- delete crd quarkssecrets.quarks.cloudfoundry.org ; \
	bazel run @kubectl//:binary -- delete crd quarksstatefulsets.quarks.cloudfoundry.org ; \
	bazel run @kubectl//:binary -- delete clusterrolebinding cf-operator-quarks-job ; \
	bazel run @kubectl//:binary -- delete clusterrolebinding cf-operator ; \
	bazel run @kubectl//:binary -- delete psp kubecf-default
```

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+	bazel query 'attr(visibility, "//visibility:public", ...)' && \
+	bazel query 'attr(visibility, "", ...)'
+
+# Installs/updates KubeCF (on whatever cluster your kubectl points to)
+apply:
+	bazel run //dev/kubecf:apply
+
+# Installs the CF operator (TODO: this should be done with bazel)
+operatorup:
+	k create namespace cf-operator && \
```
Comment (f0rmiga) [inline Makefile]:
```suggestion
	bazel run @kubectl//:binary -- create namespace cf-operator && \
```

**Comment by viovanov** [comment]
Based on discussions, this is not useful - we'll include some descriptions in the README and sample queries.

**Pull request closed without merging**